### PR TITLE
[libc++] Fixes time formatter test output for Linux on PowerPC

### PR DESCRIPTION
**Pull request by lei137**
Fix output to match actual.

### CODE REVIEW

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp [hunk 1]
```diff
@@ -821,7 +821,9 @@ static void test_valid_values_date_time() {
   // Use the global locale (fr_FR)
   check(
 // https://sourceware.org/bugzilla/show_bug.cgi?id=24054
-#if defined(__GLIBC__) && __GLIBC__ <= 2 && __GLIBC_MINOR__ < 29
```
Comment (EricWF) [inline libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:824]: Powerpc is a CPU architecture. It shouldn't affect the format of times.

Why is the output different on your platform? Is it a different  GLIBC?

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp [hunk 2]
```diff
@@ -821,7 +821,9 @@ static void test_valid_values_date_time() {
...
+#if defined(__powerpc__) && defined(__linux__)
+      SV("%c='jeu. 01 janv. 1970 00:00:00 UTC'\t%Ec='jeu. 01 janv. 1970 00:00:00 UTC'\n"),
```
Comment (EricWF) [inline libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:825]: Is the only difference here the time zone? That seems like the locale on your machine is configured differently.

**Review by EricWF — CHANGES_REQUESTED** [review]
Using `__powerpc__` to configure locale specific differences in platform is the wrong way to go about this.

You likely need to identify what libc your system has, and what version it is, and why it provides different locales.

Further, we should then figure out if those locales produce the desired output before changing our test suite to tolerate them.

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp (hunk 2 above)
Comment (lei137) [reply libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:825]: Yes, that is the only difference.  Can we update the test to be more tolerant of different configurations?

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp (hunk 1 above)
Comment (lei137) [reply libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:824]: You are right.  It's not a different GLIBC.  We just configure our locales differently.

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp (hunk 2 above)
Comment (mordante) [reply libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:825]: I think that is quite hard. The C++ code uses libc's locale. Unfortunately different vendors have different view how things are done in a specific locale. I agree this is not great, but most of our locale specific tests have a lot of `#ifdef`s.

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp (hunk 1 above)
Comment (mordante) [reply libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:824]: Is there a different way to identify the system? Or is this valid for all linux based powerpc systems, including the ancient ones?

**Comment by lei137** [comment]
@EricWF @mordante Is it possible to update the test case to NOT take into account the time zone/standard being used?  I see Apple/win32/FreeBSD does not emit that info.   I looked and we have different machines setup differently depending on where it's physically located and I am not sure the best thing to do is to switch the ones using the time standard configuration to using the GMT time zone.  Alternatively I can just disable these tests for linux PowerPC?  Your feedback is appreciated.  Thank-you!

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp (hunk 2 above)
Comment (mordante) [reply libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:825]: Actually this clock should always have UTC at timezone, this GMT is wrong too.
[[time.format]/12](http://eel.is/c++draft/time.format#12)
```
Remarks: If %Z is used, it is replaced with STATICALLY-WIDEN<charT>("UTC").
```

**Review by mordante — COMMENTED** [review]
> @EricWF @mordante Is it possible to update the test case to NOT take into account the time zone/standard being used? I see Apple/win32/FreeBSD does not emit that info. I looked and we have different machines setup differently depending on where it's physically located and I am not sure the best thing to do is to switch the ones using the time standard configuration to using the GMT time zone. Alternatively I can just disable these tests for linux PowerPC? Your feedback is appreciated. Thank-you!

@lei137 for which clocks is the time zone information wrong?

File: libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp (hunk 1 above)
Comment (lei137) [reply libcxx/test/std/time/time.syn/formatter.file_time.pass.cpp:824]: AFAIK this is the way to identify linux powerpc systems.

**Review by mordante — APPROVED** [review]
Since the CI is happy, LGTM!

**Comment by lei137** [comment]
Thank-you for identifying the issue @mordante. @EricWF Do you have any other concerns about the current fix?

**Comment by mordante** [comment]
> Thank-you for identifying the issue @mordante. @EricWF Do you have any other concerns about the current fix?

Feel free to land this, if Eric has comments we can address them post review.

**Pull request merged**